src/init.cpp

```diff
@@ -1705,7 +1699,7 @@ bool AppInitMain()
             // and because this needs to happen before any other debug.log printing
             g_logger->ShrinkDebugFile();
         }
-        if (!g_logger->OpenDebugLog()) {
+        if (!g_logger->StartLogging()) {
```

Comment:
this if needs to be moved to the exterior scope as was done in upstream 16112

Reply:
Causes test failure (see 3b38165f856b9577c5f84f3b14eac16cefa4f2a5 and https://gitlab.com/dashpay/dash/-/pipelines/315647193)

Reply:
It's probably because 16122 needs some fixing, pls see https://github.com/UdjinM6/dash/commit/2ced9fe3564ae28e81ee6392034ee95534f1ddd4

Reply:
Huh, I must've mixed patches between commits because https://github.com/bitcoin/bitcoin/pull/15849/files#diff-44fd50b51e8fc6799d38f193237fb921ec9d34306c448f64837524a17bac06eeL209 doesn't do a `scoped_lock(m_cs);` at the beginning of the function and figured it must've been redundant because of https://github.com/bitcoin/bitcoin/pull/15849/files#diff-44fd50b51e8fc6799d38f193237fb921ec9d34306c448f64837524a17bac06eeL217, it's a case of me confusing patches meant for #4190 and #4191.

Oops.